src/dialog.js

```diff
@@ -452,6 +464,146 @@ genotet.dialog.upload_ = function() {
 };
 
 /**
+ * Creates a dialog for signing up.
+ * @private
+ */
+genotet.dialog.signUp_ = function() {
+  var modal = $('#dialog');
+  modal.find('.modal-content').load(genotet.dialog.TEMPLATES_.signUp,
+    function() {
+      modal.modal();
+      var email = modal.find('#email');
+      var username = modal.find('#username');
+      var password = modal.find('#password');
+      var confirmPassword = modal.find('#confirm-password');
+      var btnSignUp = modal.find('#btn-sign-up').prop('disabled', true);
+      var inputValid = false;
+
+      var uploadReady = function() {
+        // Validate the input information.
+        inputValid = genotet.utils.validateEmail(
+            /** @type {string} */(email.val())) &&
+          genotet.utils.validateUsername(
+            /** @type {string} */(username.val())) &&
+          genotet.utils.validatePassword(
+            /** @type {string} */(password.val()));
+
+        // Checks if all required fields are filled.
+        return email.val() !== '' && username.val() !== '' &&
+          password.val() !== '' && (confirmPassword.val() == password.val() &&
+          inputValid);
+      };
+      confirmPassword.on('input', function() {
+        btnSignUp.prop('disabled', !uploadReady());
+      });
+      btnSignUp.click(function() {
+        var userInfo = {
+          type: 'sign-up',
+          email: email.val(),
+          username: username.val(),
+          password: CryptoJS.SHA256(/** @type {string} */(password.val()))
```

Comment:
You need to validate both on the front end and back end. There is no way
you can prevent a query being sent directly to the server, circumventing
whatever frontend checking you have.

On Sunday, March 20, 2016, Liana Yang <notifications@github.com> wrote:

> In src/dialog.js
> <https://github.com/ViDA-NYU/genotet/pull/145#discussion_r56759366>:
>
> > +            /** @type {string} */(password.val()));
> > +
> > +        // Checks if all required fields are filled.
> > +        return email.val() !== '' && username.val() !== '' &&
> > +          password.val() !== '' && (confirmPassword.val() == password.val() &&
> > +          inputValid);
> > +      };
> > +      confirmPassword.on('input', function() {
> > +        btnSignUp.prop('disabled', !uploadReady());
> > +      });
> > +      btnSignUp.click(function() {
> > +        var userInfo = {
> > +          type: 'sign-up',
> > +          email: email.val(),
> > +          username: username.val(),
> > +          password: CryptoJS.SHA256(/** @type {string} */(password.val()))
>
> I think the validation of password should be taken in the front end, which
> is used to remind the user to type in a validated password when he sign up
> and set up the password.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/ViDA-NYU/genotet/pull/145/files/83dd548aa3090df14c14b115859530f16cb2a2a5#r56759366>
>